analysis/02_new_model_input/04.1_topography_variables.md

```diff
@@ -0,0 +1,252 @@
+# Creating Topography Variables
+
+```python
+%load_ext jupyter_black
+from pathlib import Path
+import os
+import pandas as pd
+import geopandas as gpd
+import rasterio
+from rasterio.merge import merge
+from rasterstats import zonal_stats
+import numpy as np
+```
+
+```python
+base_url = Path(os.getenv("STORM_DATA_DIR")) / "analysis/02_new_model_input/"
+input_dir = base_url / "04_topography/input/srtm/"
+output_dir = base_url / "04_topography/output/"
+shp_output_dir = base_url / "02_housing_damage/output/"
+```
+
+```python
+grid = gpd.read_file(shp_output_dir / "phl_0.1_degree_grid_land_overlap.gpkg")
+```
+
+```python
+adm3_shp = gpd.read_file(
+    base_url / "02_housing_damage/input/phl_adminboundaries_candidate_adm3.zip"
+)
+```
+
+```python
+# reading in the files
+# merging rasters
+fileList = os.listdir(input_dir)
+mosaic_raster = []
+for file in fileList:
+    rast = rasterio.open(input_dir / file)
+    mosaic_raster.append(rast)
+
+merged_raster, out_raster = merge(mosaic_raster)
+```
+
+```python
+out_meta = rast.meta.copy()
+out_meta.update(
+    {
+        "driver": "GTiff",
+        "height": merged_raster.shape[1],
+        "width": merged_raster.shape[2],
+        "transform": out_raster,
+        "crs": grid.crs,
+    }
+)
+```
+
+```python
+with rasterio.open(input_dir / "phl_merged_srtm.tif", "w", **out_meta) as dest:
+    dest.write(merged_raster)
+```
+
+```python
+merged_rast = rasterio.open(input_dir / "phl_merged_srtm.tif")
+merged_array = merged_rast.read(1)
+```
+
+```python
+# One way of computing topography variables
+# GUI will pop up to have options for data input.
+# Compute the slope and ruggedness using it.
+# Expand Geomorphomentric Analysis
+# Select slope, enter input DEM file which is the merged file and the output file,
+# leave everything as is making sure units is degrees and click on Run.
+# Once complete, select RuggednessIndex, enter input DEM file (merged file)
+# and output file name and click Run.
+# Once complete, close the box.
+
+# whitebox.Runner()
+```
+
+```python
+# Testing out new way of computing slope and TRI.
+! gdalinfo "%STORM_DATA_DIR%/analysis/02_new_model_input/04_topography/input/srtm/phl_merged_srtm.tif"
```

Comment:
This is really cool! But I don't think this style of environment variable will work for all OS. Maybe you can get the environment variable from Python and just pass the full path name?